tket/src/Transformations/include/Transformations/ControlledGates.hpp

```diff
@@ -30,6 +30,27 @@ Circuit incrementer_borrow_1_qubit(unsigned n);
 
 Circuit incrementer_borrow_n_qubits(unsigned n);
 
+/**
+ * @brief Get an n-qubit incrementer circuit with linear depth and O(n^2) gate
+ * count. There exists a global phase difference
+ * https://arxiv.org/abs/2203.11882
+ *
+ * @param n number of qubits
+ * @param lsb set to false if we don't want to toggle the least siginificant bit
+ * @return Circuit CRx, Rx(1)
+ */
+Circuit incrementer_linear_depth(unsigned n, bool lsb = true);
+
+/**
+ * @brief Implement CnU gate with linear depth and O(n^2) gate count.
+ * https://arxiv.org/abs/2203.11882
+ *
+ * @param n number of controls
+ * @param u the controlled 2x2 unitary matrix
+ * @return Circuit CRx, Rx(1), and controlled unitary boxes
```

Comment:
And this.

Reply:
Thanks for spotting this. This one actually won't produce Rx(1) or X. It's updated now.